controllers/eventhandlers/targetgrouppolicy.go

```diff
@@ -0,0 +1,158 @@
+package eventhandlers
+
+import (
+	"context"
+	"github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gateway_api_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gateway_api "sigs.k8s.io/gateway-api/apis/v1beta1"
+	mcs_api "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+type targetGroupPolicyEventHandler struct {
+	log    gwlog.Logger
+	client client.Client
+}
+
+func NewTargetGroupPolicyEventHandler(log gwlog.Logger, client client.Client) *targetGroupPolicyEventHandler {
+	return &targetGroupPolicyEventHandler{log: log, client: client}
+}
+
+func (h *targetGroupPolicyEventHandler) getTargetRef(obj client.Object) *corev1.Service {
+	tgp := obj.(*v1alpha1.TargetGroupPolicy)
+	policyName := tgp.Namespace + "/" + tgp.Name
+
+	targetRef := tgp.Spec.TargetRef
+	if targetRef.Group != "" || targetRef.Kind != "Service" {
```

Comment:
Would it be worth separating the ```Group``` condition and the ```Kind``` condition? Given this is attached to our TargetGroupPolicy CRD, is having values we don't support even a valid use case?

Reply:
want to be clear on that we want to support this particular resource groupkind. the struct is from gwapi so we cannot enforce this by yaml validation